nifi-nar-bundles/nifi-jms-bundle/nifi-jms-processors/src/main/java/org/apache/nifi/jms/processors/JMSPublisher.java

```diff
@@ -108,6 +108,13 @@ void setMessageHeaderAndProperties(final Session session, final Message message,
                         }
                     } else {
                         // not a special attribute handled above, so send it as a property using the specified property type
+                        String key=entry.getKey();
+                        if (key.endsWith(".type")){
```

Comment:
If we go broad and apply this to anything ending in '.type' I think it deserves a WARN.